lib/pbench/server/api/resources/__init__.py

```diff
@@ -1551,15 +1556,18 @@ def _dispatch(
                     method,
                     ApiParams(body=body_params, query=query_params, uri=uri_params),
                 )
+            except APIInternalError as e:
+                self.logger.exception("{} {}", api_name, e.details)
+                abort(e.http_status, message=str(e))
             except APIAbort as e:
                 self.logger.exception("{} {}", api_name, e)
                 abort(e.http_status, message=str(e))
-            except Exception as e:
-                self.logger.exception("{} API error: {}", api_name, e)
-                abort(
-                    HTTPStatus.INTERNAL_SERVER_ERROR,
-                    message=HTTPStatus.INTERNAL_SERVER_ERROR.phrase,
-                )
+            except Exception:
+                # Construct an APIInternalError to get the UUID and standard return
+                # message.
+                x = APIInternalError("Unexpected validation exception")
+                self.logger.exception("{} {}", api_name, x.details)
+                abort(x.http_status, message=str(x))
```

Comment:
Alternately,
```
            except Exception as e:
                if not isinstance(e, (APIAbort, APIInternalError)):
                    # Construct an APIInternalError to get the UUID and standard
                    # return message, and use it instead of the actual exception
                    # (the traceback will include the actual exception).
                    e = APIInternalError("Unexpected validation exception")
                self.logger.exception(
                    "{} {}", api_name, e if isinstance(e, APIAbort) else e.details
                )
                abort(e.http_status, message=str(e))
```
And...if you actually like that (😉), consider putting it in a subroutine which we can call from other places where we do the same thing (like 1599-1610 and maybe 1587-1595).

Reply:
I didn't want to move the `abort` calls out of the main method, and it didn't seem worthwhile trying to factor out the rest.